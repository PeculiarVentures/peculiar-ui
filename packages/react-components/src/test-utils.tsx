import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
