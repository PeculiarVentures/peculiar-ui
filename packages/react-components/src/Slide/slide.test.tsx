import React from 'react';
import { renderWithWrapper as render } from '../test-utils';
import { Slide } from './index';

describe('<Slide />', () => {
  it('should render with default styles', () => {
    const {
      asFragment,
    } = render(
      <Slide>
        <div>Inside</div>
      </Slide>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  describe('directions', () => {
    const directions: React.ComponentProps<typeof Slide>['direction'][] = [
      'down',
      'left',
      'right',
      'up',
    ];

    directions.forEach((direction) => {
      describe(`direction "${direction}"`, () => {
        it('before animation', () => {
          const {
            asFragment,
          } = render(
            <Slide direction={direction}>
              <div>Inside</div>
            </Slide>,
          );

          expect(asFragment()).toMatchSnapshot();
        });

        it('after animation', () => {
          const {
            asFragment,
          } = render(
            <Slide in direction={direction}>
              <div>Inside</div>
            </Slide>,
          );

          expect(asFragment()).toMatchSnapshot();
        });
      });
    });
  });
});
