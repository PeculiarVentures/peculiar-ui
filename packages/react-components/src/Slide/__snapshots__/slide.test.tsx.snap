// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Slide /> > directions > direction "down" > after animation 1`] = `
<DocumentFragment>
  <div
    style="transform: translateY(0px); transition: transform 225ms cubic-bezier(0, 0, 0.2, 1) 0ms;"
  >
    Inside
  </div>
</DocumentFragment>
`;

exports[`<Slide /> > directions > direction "down" > before animation 1`] = `
<DocumentFragment>
  <div
    style="transform: translateY(100%); transition: transform 225ms cubic-bezier(0, 0, 0.2, 1) 0ms; visibility: hidden;"
  >
    Inside
  </div>
</DocumentFragment>
`;

exports[`<Slide /> > directions > direction "left" > after animation 1`] = `
<DocumentFragment>
  <div
    style="transform: translateX(0px); transition: transform 225ms cubic-bezier(0, 0, 0.2, 1) 0ms;"
  >
    Inside
  </div>
</DocumentFragment>
`;

exports[`<Slide /> > directions > direction "left" > before animation 1`] = `
<DocumentFragment>
  <div
    style="transform: translateX(-100%); transition: transform 225ms cubic-bezier(0, 0, 0.2, 1) 0ms; visibility: hidden;"
  >
    Inside
  </div>
</DocumentFragment>
`;

exports[`<Slide /> > directions > direction "right" > after animation 1`] = `
<DocumentFragment>
  <div
    style="transform: translateX(0px); transition: transform 225ms cubic-bezier(0, 0, 0.2, 1) 0ms;"
  >
    Inside
  </div>
</DocumentFragment>
`;

exports[`<Slide /> > directions > direction "right" > before animation 1`] = `
<DocumentFragment>
  <div
    style="transform: translateX(100%); transition: transform 225ms cubic-bezier(0, 0, 0.2, 1) 0ms; visibility: hidden;"
  >
    Inside
  </div>
</DocumentFragment>
`;

exports[`<Slide /> > directions > direction "up" > after animation 1`] = `
<DocumentFragment>
  <div
    style="transform: translateY(0px); transition: transform 225ms cubic-bezier(0, 0, 0.2, 1) 0ms;"
  >
    Inside
  </div>
</DocumentFragment>
`;

exports[`<Slide /> > directions > direction "up" > before animation 1`] = `
<DocumentFragment>
  <div
    style="transform: translateY(-100%); transition: transform 225ms cubic-bezier(0, 0, 0.2, 1) 0ms; visibility: hidden;"
  >
    Inside
  </div>
</DocumentFragment>
`;

exports[`<Slide /> > should render with default styles 1`] = `
<DocumentFragment>
  <div
    style="transform: translateX(100%); transition: transform 225ms cubic-bezier(0, 0, 0.2, 1) 0ms; visibility: hidden;"
  >
    Inside
  </div>
</DocumentFragment>
`;
